polaris-tokens/scripts/getCustomPropertyNames.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const path = require('path');
+
+const tokenGroupsDir = path.join(__dirname, '../src/token-groups');
+
+/**
+ * Allowed Polaris token custom properties.
+ *
+ * Result: ['--p-background', '--p-text', etc...]
+ */
+const getCustomPropertyNames = () => {
+  const polarisTokenCustomProperties = Array.from(
+    new Set(
+      fs
+        .readdirSync(tokenGroupsDir)
+        .map((file) => {
+          const tokenGroup = require(path.join(tokenGroupsDir, file));
+
+          return Object.keys(tokenGroup).map((token) => `--p-${token}`);
+        })
+        .flat(),
+    ),
+  );
+
+  return polarisTokenCustomProperties;
+};
```

Comment:
Can we move this to `src/utilities` and do a quick update to use `tokens`? All the `fs` and `path` logic was a hacky workaround when we were using `json` token groups.